### PR TITLE
Improvement based on PR #12 Experimental modal component

### PR DESCRIPTION
**Pull request by hachilam92**
Achieved:
1. move the "Modal" component and controlling states into "ModalContext" based on Brother Long's advice;
2. add embedded close button to "Modal";
3. a bit refactor on the "OfferButton" example, pass modal toggler to the loaded page so it can close itself;

Continuing:
1. haven't decided should we put more styles into "Modal" to reuse the style, or build extra modal-page-template components for layout and styles for flexibility, may discuss about it later in the meeting;

### CODE REVIEW

**Review by JavaScriptN0ob — APPROVED** [review]
LGTM

**Comment by Changhong-st** [comment]
LGTM

**Review by Changhong-st — APPROVED** [review]

File: src/App.js [hunk 1]
```diff
@@ -12,20 +12,24 @@ import Landing from './pages/Landing';
...
-      </Router>
+      <ModalControlProvider>
+        <Router>
+          <Switch>
+            <Route exact path="/" component={Landing} />
+            <Route path="/tasks" component={BrowseTask} />
+            <Route exact path="/create-profile" component={CreateProfile} />
+            <Route exact path="/create" component={PostTask} />
+            <PrivateRoute exact path="/profile" component={Profile} />
+            <Route path="*" component={NotFound} />
```
Comment (KieraDOG) [inline src/App.js:29]: 这里 path 可以省略

File: src/ModalContext/ModalContext.jsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  const modalControl = { 
+    toggleModal,
+    setModal, 
+  };
+
+  const closeButtonIcon = String.fromCharCode(9747);
+
+  return(
+    <ModalContext.Provider value = {modalControl}>
```
Comment (KieraDOG) [inline src/ModalContext/ModalContext.jsx:37]: 这里简单的 js style，大家review的时候都要注意一下

File: src/ModalContext/ModalContext.jsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+};
+
+const ModalContext = createContext('');
+
+export function ModalControlProvider({ children }) {
+  const [modalIsOpen, setIsOpen] = useState(false);
+  const [modalContent, setModal] = useState('');
+  
+  const toggleModal = () => {
+    setIsOpen(!modalIsOpen);
```
Comment (KieraDOG) [inline src/ModalContext/ModalContext.jsx:26]: 用 cb，不要直接依赖于 previous state vlaue

File: src/ModalContext/ModalContext.jsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    left: '50%',
+    right: 'auto',
+    bottom: 'auto',
+    borderRadius: '10px',
+    marginRight: '-50%',
+    transform: 'translate(-50%, -50%)'
+  }
+};
+
+const ModalContext = createContext('');
```
Comment (KieraDOG) [inline src/ModalContext/ModalContext.jsx:19]: 不需要传 空 string

**Pull request merged**